### PR TITLE
feat(launch): better docker image job name handling, tag -> alias

### PR DESCRIPTION
**Pull request by gtarpenning**
Fixes
-----
https://wandb.atlassian.net/browse/WB-13933

Description
-----------
Remove tag from the job name, move to alias. Launching a new docker image with the same name but different tag logs a new version, with correct aliases. 

Screen Shot 2023-06-14 at 3 52 23 PM

Screen Shot 2023-06-14 at 3 54 00 PM

### <samp>🤖 Generated by Copilot at 0d7c045</samp>

This pull request enables job artifacts to include docker image tags as part of their metadata. It updates the `job_builder.py` module to handle the `tag` attribute of image sources.

### CODE REVIEW

File: wandb/sdk/internal/job_builder.py [hunk 1]
```diff
@@ -215,7 +218,14 @@ def _build_image_job(
     ) -> Tuple[Artifact, ImageSourceDict]:
         image_name = metadata.get("docker")
         assert isinstance(image_name, str)
-        name = make_artifact_name_safe(f"job-{image_name}")
+
+        raw_image_name = image_name
+        if ":" in image_name:
+            raw_image_name, tag = image_name.split(":")
+            # TODO(gst): make this better
+            self.aliases += [tag]
```
Comment (gtarpenning) [inline wandb/sdk/internal/job_builder.py:226]: this is terrible, we instead want to communicate this to a backend process. we can put this in the metadata, and at log-time, peek into it and see if we have any tags?

**Comment by KyleGoyette** [comment]
We'll need to ship a docs PR change for this as well @gtarpenning

**Comment by gtarpenning** [comment]
https://github.com/wandb/docodile/pull/292

**Pull request merged**